File: Yakovchik_Ilya/lab2/sources/app/src/main/java/com/example/guessnumberlb2/MainActivity.java
package com.example.guessnumberlb2;

import androidx.appcompat.app.AppCompatActivity;
import android.widget.*;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    TextView tvInfo;
    EditText etInput;
    Button bControl;

    int guess;
    boolean GameFinished;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tvInfo = (TextView)findViewById(R.id.textView);
        etInput = (EditText)findViewById(R.id.EditText1);
        bControl = (Button)findViewById(R.id.button1);
        guess = (int)(Math.random()*100);
        GameFinished = false;
    }


    public void onClick(View v){
        if (!GameFinished){
            int inp=Integer.parseInt(etInput.getText().toString());
            if (inp > guess)
                tvInfo.setText(getResources().getString(R.string.ahead));
            if (inp < guess)
                tvInfo.setText(getResources().getString(R.string.behind));
            if (inp == guess)
            {
                tvInfo.setText(getResources().getString(R.string.hit));
                bControl.setText(getResources().getString(R.string.play_more));
                GameFinished = false;
                etInput.setText("");
            }
        }
        else
        {
            guess = (int)(Math.random()*100);
            bControl.setText(getResources().getString(R.string.input_value));
            tvInfo.setText(getResources().getString(R.string.try_to_guess));
            GameFinished = false;
        }

    }
}

File: Mayevsky_Alexander/lab2/sources/app/src/main/java/com/example/lab1/MainActivity.kt
package com.example.lab1

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import kotlin.math.roundToInt

class MainActivity : AppCompatActivity() {
    private var tvInfo: TextView? = null
    private var etInput: EditText? = null
    private var bControl: Button? = null
    var guess = 0
    var gameFinished = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tvInfo = findViewById(R.id.textView2)
        etInput = findViewById(R.id.editTextNumber)
        bControl = findViewById(R.id.button)
        guess = ((Math.random()*100).roundToInt())
    }

    fun onClick(view: View) {

        if (!gameFinished) {
            var text = etInput?.text?.toString()

            if(text.equals(""))
            {
                tvInfo?.text = resources.getString(R.string.emptyError)
                return
            }

            var inp = Integer.parseInt(text)
            if (inp < 0 || inp > 100){
                tvInfo?.text = resources.getString(R.string.rangeError)
                return
            }

            if (inp > guess)
                tvInfo?.text = resources.getString(R.string.ahead)

            if (inp < guess)
                tvInfo?.text = resources.getString(R.string.behind)

            if (inp == guess)
            {
                tvInfo?.text = resources.getString(R.string.hit)
                bControl?.text = resources.getString(R.string.play_more)
                gameFinished = true
            }}
        else
        {
            guess = (Math.random()*100).roundToInt()
            bControl?.text = resources.getString(R.string.input_value)
            tvInfo?.text = resources.getString(R.string.try_to_guess)
            gameFinished = false
        }

        etInput?.setText("")
    }
}